### PR TITLE
bpo-34576 warn users on security for http.server

### PR DESCRIPTION
**Pull request by fbidu**
It was proposed to add an warning for http.server regarding security
issues. The wording was provided at [bpo-26005](https://www.bugs.python.org/issue26005) by @orsenthil 

I created a section called "Security Considerations" as recommended in [Documenting Python](https://devguide.python.org/documenting/#security-considerations-and-other-concerns)
This needs to be backported to the Python 2 docs for SimpleHTTPServer but I'm not really sure how should I proceed to do that.

https://bugs.python.org/issue34576

### CODE REVIEW

File: Doc/library/http.server.rst [hunk 1]
```diff
@@ -16,6 +16,14 @@
 
 This module defines classes for implementing HTTP servers (Web servers).
 
+Security Considerations
+-----------------------
+
+http.server is meant for demo purposes and does not implement the stringent
+security checks needed of real HTTP server. We do not recommend
```
Comment (augustogoulart) [inline Doc/library/http.server.rst:23]: What do you think about adding an article on the following section?

`security checks needed of **a** real HTTP server`

**Review by orsenthil — APPROVED** [review]
LGTM

File: Doc/library/http.server.rst (hunk 1 above)
Comment (fbidu) [reply Doc/library/http.server.rst:23]: good point. But probably it should be either "security checks need for a real HTTP server" or "security checks need for real HTTP servers"

**Comment by fbidu** [comment]
@orsenthil I made a small fix in the wording of the sentence, the commit did not appear here because of the merging... Is it possible to reopen it?

**Comment by orsenthil** [comment]
@fbidu  - what is the fix? I reviewed before the merge and found the current change ok.

**Comment by orsenthil** [comment]
If anything significant,  we will have bring it in as a separate patch.

**Comment by fbidu** [comment]
@orsenthil I changed "does not implement the stringent security checks needed **of real** HTTP server" to "does not implement the stringent security checks needed **for a real** HTTP server". I think 'need for' makes more sense

**Pull request merged**